### PR TITLE
docs: Migrate Preact docs from `wmr` to `create-preact`

### PR DESCRIPTION
**Pull request by rschristian**
### Summary

[`wmr` is unfortunately no longer maintained](https://github.com/preactjs/wmr?tab=readme-ov-file#wmr). It was largely extracted into Vite and it's what we (the Preact team) recommend using in its place, [via `create-preact`](https://preactjs.com/guide/v10/getting-started#create-a-vite-powered-preact-app).

Prerendering is still functionality that's available to users, but requires a bit more work to setup than just a flag, so I moved the note on prerendering up a bit as it's an option users can enable when initializing a project. If that's what a user wants, easier to have us set it up for them than reading [the docs](https://github.com/preactjs/preset-vite?tab=readme-ov-file#prerendering-configuration) to figure out which bits they need.

### Screenshots (optional)

N/A

### Documentation checklist

- [X] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.

### CODE REVIEW

File: src/content/docs/pages/framework-guides/deploy-a-preact-site.mdx [hunk 1]
```diff
@@ -9,17 +9,25 @@ import { Render } from "~/components"
 [Preact](https://preactjs.com) is a popular, open-source framework for building modern web applications. Preact can also be used as a lightweight alternative to React because the two share the same API and component model.
 
 In this guide, you will create a new Preact application and deploy it using Cloudflare Pages.
-You will use [`wmr`](https://github.com/preactjs/wmr), an all-in-one development tool built by the Preact team, to quickly generate an optimized web application.
+You will use [`create-preact`](https://github.com/preactjs/create-preact), a lightweight project scaffolding tool to set up a new Preact app in seconds.
```
Comment (ToriLindsay) [inline src/content/docs/pages/framework-guides/deploy-a-preact-site.mdx]:
```suggestion
You will use [`create-preact`](https://github.com/preactjs/create-preact), a lightweight project scaffolding tool, to set up a new Preact app in seconds.
```

File: src/content/docs/pages/framework-guides/deploy-a-preact-site.mdx [hunk 2]
```diff
@@ -9,17 +9,25 @@ import { Render } from "~/components"
...
 ```sh
-$ npm init wmr your-project-name
+$ npm init preact
 $ cd your-project-name
 ```
 
+:::note
+
+
+**Note:** During initialization, you will notice a "Prerender app (SSG)?" option presented. With this, `create-preact` will scaffold your app to produce static HTML pages - along with their assets - for production builds, which is perfect for Pages.
```
Comment (ToriLindsay) [inline src/content/docs/pages/framework-guides/deploy-a-preact-site.mdx:26]: :::note
During initialization, select the **Prerender app (SSG)?** option to allow `create-preact` to scaffold your app to produce static HTML pages, along with their assets, for production builds. This option is perfect for Pages.
:::

File: src/content/docs/pages/framework-guides/deploy-a-preact-site.mdx [hunk 3]
```diff
@@ -9,17 +9,25 @@ import { Render } from "~/components"
...
 $ cd your-project-name
 ```
 
+:::note
+
+
+**Note:** During initialization, you will notice a "Prerender app (SSG)?" option presented. With this, `create-preact` will scaffold your app to produce static HTML pages - along with their assets - for production builds, which is perfect for Pages. 
+
+
+:::
```
Comment (ToriLindsay) [inline src/content/docs/pages/framework-guides/deploy-a-preact-site.mdx:29]:
```suggestion
```

**Review by ToriLindsay — APPROVED** [review]

**Comment by ToriLindsay** [comment]
👋Hi there, I've recreated this PR at #17226.

**Comment by rschristian** [comment]
You could've just asked me to rebase? Or rebased it yourself even, it's open to edits from contributors.

**Comment by dario-piotrowicz** [comment]
@rschristian would you like to rebase and we just reopen this PR? 🙂 

(PS: it doesn't look like I can rebase for you for some reason 😕)

**Comment by rschristian** [comment]
Sure, can do.

I'm not upset by any means, but I think it's sorta weird to copy/paste someone's work (even simple stuff like this) into another branch without even adding a `Co-authored-by: ...`

Edit: Should be rebased now

**Comment by dario-piotrowicz** [comment]
@rschristian sorry.... github doesn't actually allow me to reopen the PR anymore... I have no idea what's going on here... 😕 🤷 
Screenshot 2024-09-30 at 21 50 19
(I am guessing that a PR can't be reopen if it was force pushed or rebased when it was closed? 🤷  sorry I did not know this)

I ended up cherry-picking your commit in #17226, I hope that's fine by you 🙇 

Sorry for the whole confusion and PR recreation, I agree that it is important to credit the original author, but I can guarantee you that the mistake didn't have any ill intentions 🙂

and in any case thanks for very much for your docs contribution! 🫶

**Comment by rschristian** [comment]
>  (I am guessing that a PR can't be reopen if it was force pushed or rebased when it was closed? 🤷 sorry I did not know this)

Huh, interesting & TIL. I was wondering why the changes didn't show up here. I had no idea that'd be an issue either.

Oh well, do whatever. Important thing is just getting WMR out of there, we've (somewhat sadly) moved on to other tools.

**Pull request closed without merging**